runtime/lib/ttnn/operations/deletion/deallocate.cpp

```diff
@@ -1,16 +1,17 @@
 // SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
 //
 // SPDX-License-Identifier: Apache-2.0
-#include "dealloc.h"
+#include "deallocate.h"
 #include "tt/runtime/detail/logger.h"
 #include "tt/runtime/detail/ttnn.h"
+#include "ttnn/operations/core/core.hpp"
 
 namespace tt::runtime::ttnn::operations::deletion {
-void run(const ::tt::target::ttnn::DeallocOp *op, ProgramContext &context) {
+void run(const ::tt::target::ttnn::DeallocateOp *op, ProgramContext &context) {
   ProgramTensorPool &tensorPool = context.getTensorPool();
   ::ttnn::Tensor &tensor = tensorPool.at(op->in()->global_id());
   DEBUG_ASSERT(tensor.is_allocated());
-  tensor.deallocate();
+  ::ttnn::operations::core::deallocate(tensor, false);
```

Comment:
Does ::ttnn::deallocate not work here? Also should we be reading the force flag from the op now that we've added it to the flatbuffer schema?

Reply:
Good catch on reading the flag, will fix!

No, `::ttnn::deallocate` doesn't exist in metal :/

Reply:
Looking at the implementation it should be the same as from_device/to_device, where they also port the signature under the ttnn namespace?
```
}  // namespace core
}  // namespace operations

using operations::core::deallocate;
using operations::core::from_device;
using operations::core::reallocate;
using operations::core::squeeze_from_4D;
using operations::core::to_device;
using operations::core::unsqueeze_to_4D;

constexpr auto to_dtype = ttnn::register_operation_with_auto_launch_op<"ttnn::to_dtype", ttnn::operations::core::ToDtype>();
constexpr auto to_memory_config =
    ttnn::register_operation_with_auto_launch_op<"ttnn::to_memory_config", ttnn::operations::core::ToMemoryConfig>();
constexpr auto to_layout = ttnn::register_operation_with_auto_launch_op<"ttnn::to_layout", ttnn::operations::core::ToLayout>();

}  // namespace ttnn

```

Reply:
Oh lol, missed that! Seems noone is using it in that format, I'll change it :)